main.cpp

```diff
@@ -37,15 +27,54 @@
 #include <ibtk/muParserCartGridFunction.h>
 #include <ibtk/muParserRobinBcCoefs.h>
 
+#include <petscsys.h>
+
+#include <BergerRigoutsos.h>
+#include <CartesianGridGeometry.h>
+#include <LoadBalancer.h>
+#include <SAMRAI_config.h>
+#include <StandardTagAndInitialize.h>
+
 #include <fstream>
 #include <iostream>
+
+// Local Headers
+#include "CohesionStressRHS.h"
+
 // Function prototypes
 void output_data(Pointer<PatchHierarchy<NDIM>> patch_hierarchy,
                  Pointer<INSHierarchyIntegrator> ins_integrator,
                  const int iteration_num,
                  const double loop_time,
                  const string& data_dump_dirname);
 
+struct BetaFcn
+{
+public:
+    BetaFcn(const double eps_0, const double beta_0, const double beta_1)
+        : d_eps_0(eps_0), d_beta_0(beta_0), d_beta_1(beta_1)
+    {
+        // intentionally blank
+        return;
+    }
+    double beta(const double eps)
+    {
+        return 0.0;
```

Comment:
The actual beta function should be defined here. In lines 73-74, we're defining a wrapper that that takes a `void*`, which in actually is a pointer to an implementation of `BetaFcn`. The wrapper calls the `BetaFcn`'s implementation of `beta()`.

Reply:
So then the next item of business is to add `registerBetaFcn` to CohesionStressRHS.h/.cpp, right? And this would be a setter, I suppose. I wonder, could this be done by calling `std::bind(beta_wrapper, betaFcn)` (I'm quite certain the call is wrong here, but hopefully you understand what I'm going for here) and then assigning the class's beta function to the binding result?

Reply:
If so, my main question then is what exactly the return type of `std::bind` is (the documentation only specifies this as `A function object of unspecified type T` which doesn't exactly tell me what type I should declare CSRHS's beta function as. Ideally, I'd just use `auto` but that won't work in this case.

Reply:
In any case, I have a prototype of what I'm talking about ready, so we can go over this tomorrow

Reply:
So the way I've written things here, `CohesionStressRHS` would take a function of the type `std::function<double(double, void*)>`, along with the context `void*`. Specifically, it would have two member variables
```
std::function<double(double, void*)> d_beta_fcn;
void* d_beta_fcn_ctx;
```
Then, whenever you call `d_beta_fcn`, you would give it both the correct `double` and `d_beta_fcn_ctx`.

Reply:
If you want to use `std::bind()`, then `CohesionStressRHS` would take a function of type `std::function<double(double)>`.